sdk/messaging/azeventhubs/internal/cbs.go

```diff
@@ -23,7 +24,7 @@ const (
 )
 
 // NegotiateClaim attempts to put a token to the $cbs management endpoint to negotiate auth for the given audience
-func NegotiateClaim(ctx context.Context, audience string, conn amqpwrap.AMQPClient, provider auth.TokenProvider) error {
+func NegotiateClaim(ctx context.Context, audience string, conn amqpwrap.AMQPClient, provider auth.TokenProvider, contextWithTimeoutFn contextWithTimeoutFn) error {
```

Comment:
What's the purpose of `contextWithTimeoutFn`?

Reply:
Will do on the next push to this PR.

Reply:
(done)